pkg/collector/osm_collector.go

```diff
@@ -252,10 +253,17 @@ func (collector *OsmCollector) collectGroundTruth(meshName string) {
 		log.Print(validatingWebhookConfig)
 	}
 
-	collector.data[meshName+"_all_resources_list"] = allResourcesList
-	collector.data[meshName+"_all_resources_configs"] = allResourcesConfigs
-	collector.data[meshName+"_mutating_webhook_configurations"] = mutationWebhookConfig
-	collector.data[meshName+"_validating_webhook_configurations"] = validatingWebhookConfig
+	meshConfig, err := utils.RunCommandOnContainer("kubectl", "get", "meshconfigs", "--all-namespaces", "-o", "json")
```

Comment:
📔  Please note: for future work on this, lets make sure latter after GA of your need, we should plan a move towards using the `Client-go` packages into this collector, that will add better test-ability as well. https://github.com/kubernetes/client-go

There are some recent refactors we have done for those reason, one of the sample implementation is in ARC Pod Container work or system-err collector.

Reply:
Sure, that makes sense

Reply:
Added a test that I will also update once we refactor for client-go. Currently it's hard to test without client go.